tktermwidget/style.py

```diff
@@ -1,27 +1,275 @@
+JSON_FILE = STYLE_PATH / "styles.json"
+
+# Check the style file
+if not STYLE_PATH.exists():
+    STYLE_PATH.mkdir(parents=True)
+    with open(JSON_FILE, "w", encoding="utf-8") as f:
+        dump("{}", f)
+
+if not (JSON_FILE).exists():
+    with open(JSON_FILE, "w", encoding="utf-8") as f:
+        dump("{}", f)
+
+
+# Create style
+def write_style(**styles) -> None:
+    """Write the style into the json file"""
+    # User can use this function to write with their own style
+    # TODO: improve it, new style will overwrite the old style
+    with open(JSON_FILE, "w", encoding="utf-8") as json_obj:
+        dump(styles, json_obj, indent=1)
+
+
+# Load style
+def load_style() -> dict:
+    """Get style from the json file"""
+    # User also can use this function to get their own style
+    with open(JSON_FILE, "r", encoding="utf-8") as json_obj:
+        return load(json_obj)
+
+
+# Config style
+def config_style() -> None:
+    """Config the style with the ui"""
+    # TODO: improve the code
+    # TODO: improve the ui
+    # TODO: fix some logic bug
+    from tkinter import Text, Tk
+    from tkinter.colorchooser import askcolor
+    from tkinter.ttk import Button, Entry, Frame, Label
+
+    def selectcolor(entry: Entry, name: str) -> None:
+        """Select the color in the ui and insert into the entry and update the render"""
+        color = askcolor()[-1]
+        entry.delete(0, "end")
+        entry.insert("insert", color)
+        style[name] = color
+        renderupdate()
+
+    def renderupdate() -> None:
+        """Update the render with the latest style"""
+        render.config(
+            background=style["background"],
+            insertbackground=style["insertbackground"],
+            selectbackground=style["selectbackground"],
+            selectforeground=style["selectforeground"],
+            foreground=style["foreground"],
+        )
+        render.tag_config("select", background=style["selectbackground"], foreground=style["selectforeground"])
+        config.update()
+
+    def savestyle() -> None:
+        """Save the style"""
+        write_style(
+            background=style["background"],
+            insertbackground=style["insertbackground"],
+            selectbackground=style["selectbackground"],
+            selectforeground=style["selectforeground"],
+            foreground=style["foreground"],
+        )
+        config.destroy()
+
+    # def fillentrys() -> None:
+
+    config = Tk()
+    config.geometry("855x500")
+    config.title("Config your style")
+    config.resizable(False, False)
+
+    theme: bool = False
+    try:
+        from sv_ttk import set_theme
+    except ModuleNotFoundError:
+        pass
+    else:
+        theme = True
+        set_theme("dark")
+        config.option_add("*font", ("Cascadia Mono", 9))
+
+    style: dict[str, str, str, str, str] = DEFAULT if load_style() == "{}" else load_style()
+
+    create = Label(config, text="Create your custom style!")
+    """
```

Comment:
What is this?

Reply:
It is load style.
If there exists a user custom style, then load it, else use default style instead !